### ISSUE TITLE
Another panic during performance test

### ISSUE THREAD

**Issue by gerardsn** [nuts-foundation/nuts-node#1566]
```
2022-10-17T15:33:31.991370394Z time="2022-10-17T15:33:31Z" level=error msg="IssueVC failed" error="unable to publish the issued credential: failed to publish credential, error while creating transaction: unable to add newly created transaction to State: unable to obtain Redis transaction-level write lock: Database Error: 1 error occurred:\n\t* context deadline exceeded\n\n" module=VCR operation=IssueVC operationID=IssueVC requestURI=/internal/vcr/v2/issuer/vc user="<nil>"
2022-10-17T15:33:31.995339992Z time="2022-10-17T15:33:31Z" level=error msg="Failed to retrieve failed events: Database Error: redis: nil (notifier=private)" module=Network
2022-10-17T15:33:32.000269714Z time="2022-10-17T15:33:32Z" level=warning msg="Conn has unread data" module=Storage
2022-10-17T15:33:32.004863576Z panic: runtime error: index out of range [1] with length 1
2022-10-17T15:33:32.004889380Z
2022-10-17T15:33:32.004894960Z goroutine 105 [running]:
2022-10-17T15:33:32.004898304Z github.com/nuts-foundation/go-stoabs/redis7.shelf.visitKeys({{0xc003cbd8a0, 0xa}, {0xc0004b2950, 0xc}, {0x1a58a98, 0xc00071ad80}, {0x0, 0x0}, 0xc00053e7d0, {0x1a3b228, ...}}, ...)
2022-10-17T15:33:32.004901500Z 	/pkg/mod/github.com/nuts-foundation/go-stoabs@v1.3.0/redis7/redis.go:412 +0x388
2022-10-17T15:33:32.004904627Z github.com/nuts-foundation/go-stoabs/redis7.shelf.Iterate({{0xc003cbd8a0, 0xa}, {0xc0004b2950, 0xc}, {0x1a58a98, 0xc00071ad80}, {0x0, 0x0}, 0xc00053e7d0, {0x1a3b228, ...}}, ...)
2022-10-17T15:33:32.004907691Z 	/pkg/mod/github.com/nuts-foundation/go-stoabs@v1.3.0/redis7/redis.go:350 +0x196
2022-10-17T15:33:32.004910584Z github.com/nuts-foundation/nuts-node/network/dag.(*notifier).GetFailedEvents.func1({0x1a3b6f8, 0xc0099d88a0})
2022-10-17T15:33:32.004922392Z 	/go/network/dag/notifier.go:252 +0x98
2022-10-17T15:33:32.004925613Z github.com/nuts-foundation/go-stoabs/redis7.(*store).ReadShelf(0xc00053e7d0, {0x1a3b228?, 0xc000403200}, {0xc003cbd8a0, 0xa}, 0xc0053019b0)
2022-10-17T15:33:32.004928547Z 	/pkg/mod/github.com/nuts-foundation/go-stoabs@v1.3.0/redis7/redis.go:162 +0x139
2022-10-17T15:33:32.004931323Z github.com/nuts-foundation/nuts-node/network/dag.(*notifier).GetFailedEvents(0xc00045c980)
2022-10-17T15:33:32.004934163Z 	/go/network/dag/notifier.go:251 +0x1a9
2022-10-17T15:33:32.004936912Z github.com/nuts-foundation/nuts-node/network/dag.(*state).getFailedEvents(0xc0002c8840)
2022-10-17T15:33:32.004959298Z 	/go/network/dag/state.go:465 +0x1a6
2022-10-17T15:33:32.004963863Z github.com/nuts-foundation/nuts-node/network/dag.(*state).Diagnostics(0xc0002c8840)
2022-10-17T15:33:32.004967398Z 	/go/network/dag/state.go:478 +0x1f2
2022-10-17T15:33:32.004970686Z github.com/nuts-foundation/nuts-node/network.(*Network).collectDiagnosticsForPeers(0xc000458000)
2022-10-17T15:33:32.004973535Z 	/go/network/network.go:681 +0x75
2022-10-17T15:33:32.004981816Z github.com/nuts-foundation/nuts-node/network/transport/v2.(*peerDiagnosticsManager).start.func1()
2022-10-17T15:33:32.004985144Z 	/go/network/transport/v2/peer_diagnostics.go:56 +0x5f
2022-10-17T15:33:32.004987975Z created by github.com/nuts-foundation/nuts-node/network/transport/v2.(*peerDiagnosticsManager).start
2022-10-17T15:33:32.004990777Z 	/go/network/transport/v2/peer_diagnostics.go:49 +0xae
```
Again starts with Redis error: **"Conn has unread data"**
Also see #1543

**Comment by woutslakhorst** [comment]
I think this is an go-stoabs issue. The `keys` slice is taken directly from the command result as reference. I think the internals of the Redis lib overwrite the slice. Maybe @reinkrul can verify?

**Comment by gerardsn** [comment]
I have rerun the performance test a couple of times but have not been able to reproduce the problem

**Comment by gerardsn** [comment]
probably fixed in go-redis 9.0.0-rc2
> Bug Fixes
>
>capture error correctly in withConn ([d1bfaba](https://github.com/go-redis/redis/commit/d1bfaba549fe380d269c26cea0a0183ed1520a85))

**Comment by reinkrul** [comment]
I'd say close it for now, re-open when we see it again.
